universal-login-react/src/ui/Notifications/EmojiPlaceholders.tsx

```diff
@@ -32,12 +33,17 @@ export const EmojiPlaceholders = ({enteredCode, onEmojiClick, className}: EmojiP
     return [...emojis, ...placeholders];
   };
 
+  const isEmojiInputCompleteButInvalid = enteredCode.length === SECURITY_CODE_LENGTH && !isEmojiInputValid;
```

Comment:
Move to the same file as `isValidCode` and add unit tests.

Reply:
Done